internal/db/sql.go

```diff
@@ -162,3 +162,29 @@ SELECT sd.id, sd.day, sd.opens_at, sd.closes_at, sd.open_time, sd.open_day, sd.c
 FROM public.schedule_days sd
 WHERE sd.schedule_id = $1
 `
+
+const contentCurationData = `
+SELECT s.id as service_id, s.name as service_name, s.resource_id, r.name, r.website, s.email, s.updated_at
+FROM services s
+LEFT JOIN resources r
+ON s.resource_id = r.id
+WHERE (s.id in (select distinct a.service_id from categories_services a
+left join services b
+on a.service_id = b.id
+where a.category_id in (select id from categories where name in ('Food pantries', 'Daily free meals for all', 'Food for children', 'Home-delivered groceries and home-delivered meals','Congregate meals for seniors and people with disabilities', 'Food benefits', 'Emergency Food','Coronavirus (COVID-19) Testing', 'Coronavirus-Related Urgent Care', 'Other Medical Services', 'Mental Health Urgent Care', 'Other Mental Health Services','Showers', 'Laundry', 'Clothing', 'Diaper Bank', 'Hygiene kits', 'Portable Toilets and Hand-Washing Stations', 'I am under 18 years old.', 'I am between 18 and 24 years old.', 'I am over 24 years old without children.', 'We are a family with children under 18 years old.', 'I am experiencing homelessness and I need immediate help finding shelter.', 'I am experiencing homelessness (on the street, couchsurfing, or other) and I need long-term housing assistance.', 'I am not currently experiencing homelessness, but I am looking for a long-term affordable housing unit.', 'My landlord gave me an eviction notice and I need legal help', 'My landlord told me I would get evicted and I need advice', 'I have not been able to pay my rent and I do not know what to do', 'I am not getting along with my neighbor(s) and /or my landlord and I need advice', 'Emergency Financial Assistance', 'Financial assistance for living expenses', 'Unemployment Insurance-based Benefit Payments', 'Job Placement Support', 'Vocational Training Programs', 'Job Board', 'Housing Assistance', 'Legal Assistance', 'Youth Services', 'Counseling Assistance', 'General Help', 'Temporary Shelter for Women', 'Transitional Housing for Women', 'Domestic Violence Counseling', 'Residential Treatment', 'Healthier Habits/Safer Use', 'Pregnancy Wraparound Services', 'Meetings/Support', 'Safe place to sober up', 'Reward Program', 'Medication Treatment', 'Computer or Internet Access', 'Covid-internet', 'Smartphones', 'Help Pay for Internet or Phone', 'Technology'))
```

Comment:
I know you wrote that you wanted to refactor this query, but I was wondering what approach we want to take for all these category names. Hardcoding the names isn't great, since we sometimes tweak the names, as they're user-facing. However, replacing them with their primary key IDs would make this query basically unreadable, not to mention I don't think we have stable and consistent IDs between prod and development environments.

In the short term, could we at least add a sanity check that this hardcoded list of category names exactly matches the DB, that is, check that every single one of these matches exactly one category in the DB? I think that could be done by performing a query to fetch all categories matching a name in this list and asserting that the length of the search result matches the length of this list.

Longer-term, perhaps all this data should actually be captured in the DB, and we could build a UI so that the content curation team could manage which categories should be used in these queries?

Reply:
Yea I need to see how we decided what this list is and see if we can make it dynamic and represented by some database state.